File: 1534-统计好三元组.py
class Solution:
    def countGoodTriplets(self, arr: List[int], a: int, b: int, c: int) -> int:
        lens = len(arr)
        ans = 0
        for i in range(lens-2):
            for j in range(i+1, lens-1):
                if abs(arr[i] - arr[j]) <= a:
                    for k in range(j+1, lens):
                        if abs(arr[j] - arr[k]) <= b and abs(arr[i] - arr[k]) <= c:
                            ans+=1
        return ans

File: 1534-统计好三元组.cpp
class Solution {
public:
    int countGoodTriplets(vector<int>& arr, int a, int b, int c) {
        int ans=0;
        int lens=arr.size();
        for(int i=0;i<lens-2;i++)
        {
            for(int j=i+1;j<lens-1;j++)
            {
                if(abs(arr[i]-arr[j])<=a)
                {
                    for(int k=j+1;k<lens;k++)
                    {
                        if(abs(arr[j]-arr[k])<=b && abs(arr[i]-arr[k])<=c)
                            ans++;
                    }
                }
            }
        }
        return ans;
    }
};
